src/oop/object_interactions/medium/q6/MainTest.java

```diff
@@ -0,0 +1,197 @@
+                    new Optional(new StringLiteral(" "))
+            });
+            _assertEquals(pass + move, testClass.getFieldValue(classInstance, numPassengersAttributeName),
+                    "Your " + transferPassengersMethodName + " method does not correctly add the " + numPassengersAttributeName + " attribute of the recipient.");
+            _assertEquals(busyPass - move, busy.getNumPassengers(),
+                    "Your " + transferPassengersMethodName + " method does not correctly decrease the " + numPassengersAttributeName + " attribute of the donor.");
+        } else {
+            output = testClass.callMethod(transferPassengersMethodName, methodArguments, new String[]{"public"}, classInstance, new Clause[]{
+                    new StringLiteral("This plane is full too and cannot accommodate anymore passengers"),
+                    new Optional(new StringLiteral(" "))
+            });
+            _assertEquals(pass, testClass.getFieldValue(classInstance, numPassengersAttributeName),
+                    "Your " + transferPassengersMethodName + " method does not correctly add the " + numPassengersAttributeName + " attribute of the recipient.");
+            _assertEquals(busyPass, busy.getNumPassengers(),
+                    "Your " + transferPassengersMethodName + " method does not correctly decrease the " + numPassengersAttributeName + " attribute of the donor.");
+            _assertEquals(seat, testClass.getFieldValue(classInstance, maxSeatsAttributeName),
+                    "Your " + transferPassengersMethodName + " method does not correctly add the " + maxSeatsAttributeName + " attribute of the recipient.");
+            _assertEquals(busySeat, busy.getMaxSeats(),
+                    "Your " + transferPassengersMethodName + " method does not correctly decrease the " + numPassengersAttributeName + " attribute of the donor.");
+        }
+        assertNull(output, String.join(" ", "Your", transferPassengersMethodName, "should not return any output"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void airplaneClassHasCorrectToStringMethod(int seat, int pass) throws Throwable {
+        Object[][] arguments = {
+                {seat, int.class},
+                {pass, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String[] methodModifiers = {"public"};
+        String expected = pass + " out of  " + seat + " seats are filled";
+        String incorrectToStringMessage = String.join(" ",
+                "Your", className, " toString method does not return the correct String.");
+        Object output = testClass.callMethod("toString", methodModifiers, classInstance);
+        _assertEquals(expected, output, incorrectToStringMessage);
+    }
+
+    @Test
+    public void correctMainMethod() throws Throwable {
+        Clause[] clauses = {
+                new StringLiteral("We moved 2 passengers"),
```

Comment:
Consider making the numbers present in the main method output variables so they can be varied on website entry